config/nginx/dice-staging.tripleawarclub.org

```diff
@@ -1,14 +1,33 @@
 server {
   listen 80;
   listen [::]:80;
+  listen 443 ssl http2;
+  listen [::]:443 ssl http2;
   server_name dice-staging.tripleawarclub.org;
   root /usr/share/nginx/html/dice-staging.tripleawarclub.org/public_html;
 
+  ssl_certificate /etc/letsencrypt/live/tripleawarclub.org/fullchain.pem;
+  ssl_certificate_key /etc/letsencrypt/live/tripleawarclub.org/privkey.pem;
+
+  # Turn on OCSP stapling as recommended at
+  # https://community.letsencrypt.org/t/integration-guide/13123
+  # requires nginx version >= 1.3.7
+  ssl_stapling on;
+  ssl_stapling_verify on;
+  add_header Strict-Transport-Security "max-age=31536000";
+  ssl_ciphers 'EECDH+AESGCM:EDH+AESGCM:AES256+EECDH:AES256+EDH';
+  ssl_prefer_server_ciphers on;
+  ssl_dhparam /etc/dhparam/dhparams.pem;
+
   access_log /var/log/nginx/dice-staging.tripleawarclub.org-access.log;
   error_log /var/log/nginx/dice-staging.tripleawarclub.org-error.log;
 
   index index.php;
 
+  location / {
+    try_files $uri $uri/ =404;
+  }
```

Comment:
@RoiEXLab Is this necessary to serve the _.well-known_ directory?

Reply:
Potentially. I'm not sure though, because it had worked before.
You might want to check using the certbot dry run feature and the renew command.

Reply:
I.e.:`sudo certbot renew --dry-run`
Which validates the domain, but doesn't replace the certificate on success

Reply:
Is there a way to test it without having to deploy the configuration change first?  I guess I could run that command on the production domain (where this location rule is already present) and look at the logs to see what HTTP accesses were made.

> because it had worked before

So you're saying you tested TLS on the `dice-staging` subdomain when this rule wasn't present?  I just copied this rule from the production configuration because I thought it was required.  How about if I just remove it from this PR and see if everything works?  If not, we can add it in a follow-up PR.  At this time, HTTPS isn't supported, so nothing will be technically broken if this PR doesn't correctly enable TLS.

Reply:
Per [this reference](https://letsencrypt.org/how-it-works/), `certbot` needs to either

* provision a DNS record under our domain, or
* provision an HTTP resource under a well-known URI on the web server.

So, if it's the former, we don't need this rule, but if the latter, then we do.

Is there a way to determine which method our `certbot` is configured to use?  Or do we not have a choice in the matter and have to be able to support both methods?

Reply:
@ssoloff 
> So you're saying you tested TLS on the dice-staging subdomain when this rule wasn't present?

Otherwise I wouldn't have been able to extend the certificate for the dice-staging subdomain.
I didn't change anything with the DNS config, so it must be the latter case.
As far as I'm aware of, the cerbot tool creates a file under a given webroot, and checks if this file can be accessed publicly.
So while it looks like this rule is needed I'd assume there is some default nginx behaviour that simply serves files if a webroot setting is present?

Reply:
> So while it looks like this rule is needed I'd assume there is some default nginx behaviour that simply serves files if a webroot setting is present?

:+1: That sounds reasonable.  I will go ahead and remove the rule for now.  We can always add it later if we run into problems.

Reply:
Don't know why I didn't just try this before, but...  I created a file on the staging server under _/.well-known_ and verified I could GET it using the current configuration.  So that validates the theory that there is some default rule in Nginx to handle this.